test/upgradability/timelockController.spec.ts

```diff
@@ -0,0 +1,309 @@
+import * as t from "../../types/truffle-contracts/index";
+import { TestEngine } from "../../test-helpers/TestEngine";
+
+const { BN, time } = require("@openzeppelin/test-helpers");
+
+var chai = require("chai");
+var expect = chai.expect;
+
+const AccountTokenLib = artifacts.require("AccountTokenLib");
+const Accounts: t.AccountsContract = artifacts.require("Accounts");
+const Bank: t.BankContract = artifacts.require("Bank");
+const SavingAccount: t.SavingAccountContract = artifacts.require("SavingAccount");
+const ProxyAdmin: t.ProxyAdminContract = artifacts.require("ProxyAdmin");
+const TimelockController: t.TimelockControllerContract = artifacts.require("TimelockController");
+
+const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+const ZERO = new BN(0);
+const ONE_MIN = new BN(60); // 60 sec
+const ONE_HOUR = new BN(60).mul(ONE_MIN); // 60 mins
+
+contract("TimelockController for ProxyAdmin", async (accounts) => {
+    let owner = accounts[0];
+    let testEngine: TestEngine;
+    let savingAccount: t.SavingAccountWithControllerInstance;
+
+    before(function () {
+        this.timeout(0);
+        testEngine = new TestEngine();
+    });
+
+    beforeEach(async function () {
+        this.timeout(0);
+        savingAccount = await testEngine.deploySavingAccount();
+    });
+
+    it("should have all proxy contract", async () => {
```

Comment:
I think we should add `should fail` test cases as well to ensure that it doesn't allow upgrades when the timelock is imposed and instead gets reverted.

Reply:
added a failure test case when delay is not over

Reply:
Should be fine then, I was thinking if there could be more cases

Reply:
we added basic tests to ensure that our required features do work.

Reply:
yeah right, those look okay 👍